File: tests/test5.cpp
// template file
#include <fstream>
#include <iostream>
#include <vector>
#include <string>
#include "json.hpp"

using namespace std;
using json = nlohmann::json;

// insert Solution class here
class Solution {
public:
    string longestPalindrome(string s) {
        
        int n = s.length();
        
        int mxlen = 0, start = 0;
        
        for (int i = 0; i < n; i++) {
            
            int j = i;
            int k = i;
            
            while (k+1 < n && s[k] == s[k+1]) k++;
            
            while (j-1 >= 0 && k+1 < n && s[j-1] == s[k+1]) {
                j--; k++;
            }
            
            if (mxlen < k-j+1) {
                mxlen = k-j+1;
                start = j;
            }
        }
        
        return s.substr(start, mxlen);
    }
};

int main() {
    Solution sol;

    ifstream i("tc5.json");
    json j;
    i >> j;

    vector<string> vecInput = j["input"];
    vector<vector<string>> vecExpected = j["expected"];

    bool isOk = true;
    // test
    for (size_t i = 0; i < vecInput.size(); i++) {
        string solution  = sol.longestPalindrome(vecInput.at(i));
        vector<string> expected = vecExpected.at(i);
        if (find(expected.begin(), expected.end(), solution) == expected.end()) {
            isOk = false;
            break;
        }
    }
    if (isOk) {
        json output = {
            {"result", "OK"}
        };
        cout << output.dump(4) << endl;
    }
    cout << "test completed" << endl;
    i.close();
}

File: tests/test5.py
# template file
import json
from typing import List


# insert Solution class here
class Solution:
    def longestPalindrome(self, s):
        lenS = len(s)
        if lenS <= 1: return s
        minStart, maxLen, i = 0, 1, 0
        while i < lenS:
            if lenS - i <= maxLen / 2: break
            j, k = i, i
            while k < lenS - 1 and s[k] == s[k + 1]: k += 1
            i = k + 1
            while k < lenS - 1 and j and s[k + 1] == s[j - 1]:  k, j = k + 1, j - 1
            if k - j + 1 > maxLen: minStart, maxLen = j, k - j + 1
        return s[minStart: minStart + maxLen]


with open("tc5.json", "r") as read_file:
    data = json.load(read_file)

sol = Solution()
isOk = True

listInput = data["input"]
listExpected = data["expected"]

for i, element in enumerate(listInput):
    expected = listExpected[i]
    solution = sol.longestPalindrome(element)
    if solution not in expected:
        isOk = False
        break

if isOk:
    output = {
        "result": "OK"
    }
    json_object = json.dumps(output, indent=4)
    print(json_object)

print("test completed")